File: services/messaging-service/src/main/java/com/project/messaging/model/Message.java
package com.project.messaging.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "messages")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Message {

    @Id
    private UUID id;

    private UUID conversationId;

    private String senderId;   // Keycloak ID

    @Column(columnDefinition = "TEXT")
    private String content;

    private String messageType;

    private Boolean isRead;

    private LocalDateTime createdAt;
}

File: legal-platform-web/src/types/message.ts
export interface Message {
  id: string;
  conversationId: string;
  senderId: string;
  content: string;
  messageType: string;
  isRead: boolean;
  createdAt: string;
}

export interface Conversation {
  id: string;
  clientId: string;
  lawyerId: string;
  appointmentId: number;
  documentId?: string;
  enabled: boolean;
  createdAt: string;
}

export interface SendMessageRequest {
  conversationId: string;
  content: string;
}

export interface ChatMessageRequest {
  conversationId: string;
  content: string;
}
